src/sb/app/project/data.cljc

```diff
@@ -120,24 +121,51 @@
       :entity/project-fields))
 
 (q/defx new!
-  {:prepare [az/with-account-id!]}
-  [{:keys [account-id]} project]
+  {:prepare [az/with-account-id!
+             (az/with-member-id! (comp :entity/parent :project))]}
+  [{:keys [account-id member-id project]}]
   ;; TODO
   ;; verify that user is allowed to create a new project in parent
+  ;; check if project creation can even be restricted for board members
   (let [project    (dl/new-entity project :project :by account-id)
-        membership (member.data/new-entity-with-membership project
-                                                           (->> (:entity/parent project)
-                                                                (iterate (comp :entity/parent dl/entity))
-                                                                (take-while identity)
-                                                                (some (partial az/membership-id account-id)))
-                                                           #{:role/project-admin})]
+        membership (member.data/new-entity-with-membership project member-id #{:role/project-admin})]
     (validate/assert project :project/as-map)
+    (validate/assert (update membership :membership/entity sch/wrap-id) :membership/as-map)
     (db/transact! [membership])
     {:entity/id (:entity/id project)}))
 
 (q/defx delete!
   "Mutation fn. Marks project as deleted by given project-id."
-  [_req {:keys [project-id]}]
-  ;; TODO: auth
-  (db/transact! [[:db/add [:entity/id project-id] :entity/deleted-at (java.util.Date.)]])
+  {:prepare [az/with-account-id!]}
+  [{:keys [project-id account-id]}]
+  (az/auth-guard! (az/editor-role? (az/all-roles account-id (dl/entity project-id)))
+      "Not authorized to delete"
+    (db/transact! [[:db/add [:entity/id project-id] :entity/deleted-at (java.util.Date.)]])
+    {:body ""}))
```

Comment:
should we consider returning some tx data for these effects, so that the client receives the `[[:entity/id project-id] :entity/deleted-at <date>]` triple immediately? (ofc it will also see updated queries but for simple effects it may make sense to return the affected triples?)